pkg/edgeview/src/copyfile.go

```diff
@@ -184,18 +184,24 @@ func recvCopyFile(msg []byte, fstatus *fileCopyStatus, mtype int) {
 		fstatus.buf = make([]byte, info.Size)
 
 		fmt.Printf("file: name %s, size %d\n", fstatus.filename, fstatus.fileSize)
-		fstatus.gotFileInfo = true
 
 		_, err = os.Stat(fileCopyDir)
 		if err != nil {
 			sendCopyDone("file stat ", err)
 			return
 		}
-		fstatus.f, err = os.Create(fileCopyDir + fstatus.filename)
+		filePath := filepath.Join(fileCopyDir, fstatus.filename)
+		// check if we receive valid filename to avoid touch of files outside fileCopyDir
+		if !strings.HasPrefix(filePath, fileCopyDir) {
```

Comment:
I guess this check still can be fooled by `/path_allowed_to_access/../no_no_no_you_should_not_be_here`

Reply:
Join function do cleanup of the path inside.

Reply:
oh, that's neat